### PR TITLE
Create the cloud-governance infra: User, Policy, Bucket

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [ ] enhancement
- [x] documentation
- [ ] dependencies

## Description

Create the CloudGovernance Infra:
- User
- Policy 
- Bucket
- AccessKey

Created the terraform script to create the resources.

Note: Access Keys will be displayed on the terminal output, copied and shared with the user.

### CODE REVIEW

**Comment by ebattat** [comment]
@athiruma, do u want to add how_to image ?

**Comment by athiruma** [comment]
> @athiruma, do u want to add how_to image ?

Writing README docs

**Comment by athiruma** [comment]
> > @athiruma, do u want to add how_to image ?
> 
> Writing README docs

Done, please verify the docs

File: iam/clouds/aws/CloudGovernanceInfra/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+```shell
+./create_infra.sh --username "$USERNAME" --s3-bucket-name "$BUCKET_NAME" --policy-type delete
+```
+
+- To get help
+
+```shell
+./create_infra.sh --help
+```
```
Comment (ebattat) [inline iam/clouds/aws/CloudGovernanceInfra/README.md:80]: Do you want to add image that the explain the process ?

File: jenkins/tenant/aws/README.md [hunk 2]
```diff
@@ -1,12 +1,15 @@
 # How to run cloud-governance on Tenant Accounts
 
 Steps
-1. Create AWS User and attach user by [CloudGovernanceDeletePolicy.json](../../../iam/clouds/aws/CloudGovernanceDeletePolicy.json). [ Note: Replace account_id with actual account id]
+
+1. Create AWS User and attach user
+   by [CloudGovernanceDeletePolicy.json](../../../iam/clouds/aws/CloudGovernanceCloudCreds/CloudGovernanceDeletePolicy.json). [ Note: Replace account_id with actual account id]
```
Comment (ebattat) [inline jenkins/tenant/aws/README.md:6]: Why its different from the create infra flow ?

File: jenkins/tenant/aws/README.md [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
-2. Create S3 bucket
-3. Add kind secret-text to jenkins with below naming conventions
-   1. ${account_name}-aws-access-key-id
-   2. ${account_name}-aws-secret-key-id
-   3. ${account_name}-s3-bucket
-4. Create folder named that you want to run the cloud-governance policies and copy the file in templates.
-5. Add account_name to account variable in this [PolicyJenkinsfileDaily](../aws/template/PolicyJenkinsfileDaily) and [TaggingJenkinsfileHourly](../aws/template/TaggingJenkinsfileHourly).
-6. Create two Jenkins jobs by using this two Jenkinsfile
+
+1. Create IAM User with Read/Delete Permissions and create S3 bucket.
```
Comment (ebattat) [inline jenkins/tenant/aws/README.md:5]: I think you need to update the readme to use terraform steps
Also pls add the tfvars file for variable

File: iam/clouds/aws/CloudGovernanceInfra/README.md (hunk 1 above)
Comment (ebattat) [reply iam/clouds/aws/CloudGovernanceInfra/README.md:41]: ./create_infra.sh --username "$USERNAME" --s3-bucket-name "$BUCKET_NAME" --policy-type delete
Dont need to run 
terraform destroy

File: iam/clouds/aws/CloudGovernanceInfra/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+```shell
+curl -L https://github.com/redhat-performance/cloud-governance/raw/main/iam/clouds/aws/CloudGovernanceInfra.tar | tar -xzvf -
+```
+
+- Create CloudGovernance Infra: User, Policy and Bucket
+
+```shell
+terraform init
+terraform apply -var=IAM_USERNAME="${USERNAME}" -var=IAM_POLICY_NAME="${IAM_POLICY_NAME}" -var =S3_BUCKET_NAME="${S3_BUCKET_NAME}"
```
Comment (ebattat) [inline iam/clouds/aws/CloudGovernanceInfra/README.md:26]: Pls put the value of 
USERNAME
IAM_POLICY_NAME
S3_BUCKET_NAME

**Comment by athiruma** [comment]
/review @ebattat 
Added the module structure

**Comment by ebattat** [comment]
@athiruma, pls remove all the readme file in module and have only 1 readme with working steps:
1. Deploy S3 bucket (once for logs)
2. Deploy IAM read role (dry_run==yes)
3. Deploy IAM delete role(dry_run==no => actions)

**Comment by athiruma** [comment]
> @athiruma, pls remove all the readme file in module and have only 1 readme with working steps:
> 
> 1. Deploy S3 bucket (once for logs)
> 2. Deploy IAM read role (dry_run==yes)
> 3. Deploy IAM delete role(dry_run==no => actions)

done

File: iam/clouds/aws/CloudGovernanceInfra/CloudGovernanceReadPolicy.json [hunk 5]
```diff
@@ -6,14 +6,21 @@
...
+        "resource-explorer-2:ListViews",
+        "resource-explorer-2:Search"
       ],
       "Resource": "*"
     },
     {
       "Sid": "EC2AccountLevel",
       "Effect": "Allow",
       "Action": [
+        "ec2:DeleteTags",
```
Comment (ebattat) [inline iam/clouds/aws/CloudGovernanceInfra/CloudGovernanceReadPolicy.json:23]: Do we need it in Read policy: ec2:DeleteTags

File: iam/clouds/aws/CloudGovernanceInfra/CloudGovernanceReadPolicy.json [hunk 6]
```diff
@@ -70,7 +76,8 @@
       "Action": [
         "elasticloadbalancing:DescribeTags",
         "elasticloadbalancing:AddTags",
-        "elasticloadbalancing:DescribeLoadBalancers"
+        "elasticloadbalancing:DescribeLoadBalancers",
+        "elasticloadbalancing:RemoveTags"
```
Comment (ebattat) [inline iam/clouds/aws/CloudGovernanceInfra/CloudGovernanceReadPolicy.json:80]: "elasticloadbalancing:RemoveTags" read policy ?

**Pull request merged**